FWCore/Framework/interface/CallbackExternalWork.h

```diff
@@ -0,0 +1,199 @@
+              typename TRecord,       //the record passed in as an argument
+              typename TDecorator     //allows customization using pre/post calls
+              = CallbackSimpleDecorator<TRecord>>
+    class CallbackExternalWork
+        : public CallbackBase<CallbackExternalWork<T, TAcquireFunc, TProduceFunc, TReturn, TRecord, TDecorator>,
+                              T,
+                              TProduceFunc,
+                              TReturn,
+                              TRecord,
+                              TDecorator> {
+    public:
+      using Base = CallbackBase<CallbackExternalWork, T, TProduceFunc, TReturn, TRecord, TDecorator>;
+
+      CallbackExternalWork(T* iProd, TProduceFunc iProduceFunc, unsigned int iID, const TDecorator& iDec = TDecorator())
+          : Base(iProd, std::make_shared<TProduceFunc>(std::move(iProduceFunc)), iID, iDec),
+            acquireFunction_(std::shared_ptr<TAcquireFunc>()) {}
+
+      CallbackExternalWork(T* iProd,
+                           TAcquireFunc iAcquireFunc,
+                           TProduceFunc iProduceFunc,
+                           unsigned int iID,
+                           const TDecorator& iDec = TDecorator())
+          : Base(iProd, std::make_shared<TProduceFunc>(std::move(iProduceFunc)), iID, iDec),
+            acquireFunction_(std::make_shared<TAcquireFunc>(std::move(iAcquireFunc))) {}
+
+      CallbackExternalWork(T* iProd,
+                           std::shared_ptr<TAcquireFunc> iAcquireFunc,
+                           std::shared_ptr<TProduceFunc> iProduceFunc,
+                           unsigned int iID,
+                           const TDecorator& iDec = TDecorator())
+          : Base(iProd, std::move(iProduceFunc), iID, iDec), acquireFunction_(std::move(iAcquireFunc)) {}
+
+      CallbackExternalWork* clone() {
+        return new CallbackExternalWork(
+            Base::producer(), acquireFunction_, Base::produceFunction(), Base::transitionID(), Base::decorator());
+      }
+
+      WaitingTaskHolder makeRunModuleTaskHolderImpl(oneapi::tbb::task_group* group,
+                                                    ServiceWeakToken const& token,
+                                                    EventSetupRecordImpl const* eventSetupRecordImpl,
+                                                    EventSetupImpl const* eventSetupImpl) {
+        bool emitPostPrefetchingSignal = true;
+        if (acquireFunction_) {
+          emitPostPrefetchingSignal = false;
+        }
+        WaitingTask* runModuleTask =
+            Base::makeProduceTask(group, token, eventSetupRecordImpl, eventSetupImpl, emitPostPrefetchingSignal);
+
+        if (acquireFunction_) {
+          WaitingTaskWithArenaHolder waitingTaskWithArenaHolder(*group, makeExceptionHandlerTask(runModuleTask, group));
+          runModuleTask = makeAcquireTask(
+              std::move(waitingTaskWithArenaHolder), group, token, eventSetupRecordImpl, eventSetupImpl);
+        }
```

Comment:
I think these tasks could also be set up under the assumption that `acquireFunction_` is always set.

Reply:
Done.